test/extensions/filters/http/compressor/compressor_filter_speed_test.cc

```diff
@@ -192,102 +256,265 @@ compressChunks8192/5/manual_time        15.9 ms         16.1 ms           45
+  const auto idx = state.range(0);
+  const auto& params = gzip_compression_params[idx];
+
+  for (auto _ : state) { // NOLINT
+    std::vector<Buffer::OwnedImpl> chunks = generateChunks(30, 4096);
+    compressWith(CompressorLibs::Gzip, std::move(chunks), params, decoder_callbacks, state);
+  }
+}
+BENCHMARK(compressChunks4096WithGzip)
+    ->DenseRange(0, 8, 1)
+    ->UseManualTime()
+    ->Unit(benchmark::kMillisecond);
+
+// NOLINTNEXTLINE(readability-identifier-naming)
+static void compressChunks1024WithGzip(benchmark::State& state) {
+  NiceMock<Http::MockStreamDecoderFilterCallbacks> decoder_callbacks;
+  const auto idx = state.range(0);
+  const auto& params = gzip_compression_params[idx];
+
+  for (auto _ : state) { // NOLINT
+    std::vector<Buffer::OwnedImpl> chunks = generateChunks(120, 1024);
+    compressWith(CompressorLibs::Gzip, std::move(chunks), params, decoder_callbacks, state);
+  }
+}
+BENCHMARK(compressChunks1024WithGzip)
+    ->DenseRange(0, 8, 1)
+    ->UseManualTime()
+    ->Unit(benchmark::kMillisecond);
+
+static std::vector<CompressionParams> zstd_compression_params = {
+    // level1 + default
+    {1, 0, 0, 0},
+
+    // level2 + default
+    {2, 0, 0, 0},
+
+    // level3 + default
+    {3, 0, 0, 0},
+
+    // level4 + default
+    {4, 0, 0, 0},
+
+    // level5 + default
+    {5, 0, 0, 0},
+
+    // level6 + default
+    {6, 0, 0, 0},
+
+    // level7 + default
+    {7, 0, 0, 0},
+
+    // level1 + default
+    {8, 0, 0, 0},
+
+    // level8 + default
+    {9, 0, 0, 0},
+
+    // level9 + default
+    {10, 0, 0, 0},
+
+    // level10 + default
+    {1, 0, 0, 0},
+
+    // level11 + default
+    {11, 0, 0, 0},
+
+    // level12 + default
+    {12, 0, 0, 0},
+
+    // level13 + default
+    {13, 0, 0, 0},
+
+    // level14 + default
+    {14, 0, 0, 0},
+
+    // level15 + default
+    {15, 0, 0, 0},
+
+    // level16 + default
+    {16, 0, 0, 0},
+
+    // level17 + default
+    {17, 0, 0, 0},
+
+    // level18 + default
+    {18, 0, 0, 0},
+
+    // level19 + default
+    {19, 0, 0, 0},
+
+    // level20 + default
+    {20, 0, 0, 0},
+
+    // level21 + default
+    {21, 0, 0, 0},
+
+    // level22 + default
+    {22, 0, 0, 0}};
 
-static void compressFull(benchmark::State& state) {
+// NOLINTNEXTLINE(readability-identifier-naming)
+static void compressFullWithZstd(benchmark::State& state) {
   NiceMock<Http::MockStreamDecoderFilterCallbacks> decoder_callbacks;
   const auto idx = state.range(0);
-  const auto& params = compression_params[idx];
+  const auto& params = zstd_compression_params[idx];
+  Envoy::Extensions::HttpFilters::Compressor::Result res;
 
-  for (auto _ : state) {
+  for (auto _ : state) { // NOLINT
     std::vector<Buffer::OwnedImpl> chunks = generateChunks(1, 122880);
-    compressWith(std::move(chunks), params, decoder_callbacks, state);
+    res = compressWith(CompressorLibs::Zstd, std::move(chunks), params, decoder_callbacks, state);
```

Comment:
in the gzip benchmarks we don't keep the result, what's the reason we do for zstd?

Reply:
sry, I forget to remove it, thanks.